src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1901,9 +1927,39 @@ private void doGenKeyPair(String alias, String dname, String keyAlgName,
             throw new Exception(form.format(source));
         }
 
-        CertAndKeyGen keypair =
-                new CertAndKeyGen(keyAlgName, sigAlgName, providerName);
+        if (reqSigner && signerAlias == null) {
+            MessageFormat form = new MessageFormat(rb.getString
+                    ("The.signer.option.must.be.specified.for.the.key.algorithm.keyAlgName"));
+            Object[] source = {keyAlgName};
+            throw new Exception(form.format(source));
+        }
+
+        CertAndKeyGen keypair;
+        if (signerAlias != null) {
+            signerFlag = true;
+
+            if (keyStore.containsAlias(signerAlias) == false) {
```

Comment:
It's probably more precise to make sure the entry is a `PrivateKeyEntry` because we have other entries like `TrustedCertificateEntry` and `SecretKeyEntry`. Or you can double check this below to ensure both `signerPrivateKey` and `signerCert` are non null.

Reply:
As `RecoveryKey()` will make sure if the entry exists in the keystore and is a `PrivateKeyEntry`, removed this checking and updated to check for if `signerCert` is null.

Reply:
Yes, it must be a private key entry. On the other hand, I think it's unnecessary to check about the `signerCert`. I don't think it'll be ever null.

Reply:
Ok, removed this checking.